Source/WebCore/bindings/js/JSDOMConvertStrings.cpp

```diff
@@ -118,4 +122,39 @@ AtomString valueToUSVAtomString(JSGlobalObject& lexicalGlobalObject, JSValue val
     return replaceUnpairedSurrogatesWithReplacementCharacter(WTFMove(string));
 }
 
+// https://w3c.github.io/trusted-types/dist/spec/#get-trusted-type-compliant-string-algorithm
```

Comment:
This is a slightly tricky one because the algorithm as specced is split across two functions within WebKit. One here which is largely working with IDL types and then the one in TrustedTypeUtils which is where most of the actual algorithm is implemented.

So it's not really possible to follow it line by line. I have however renamed value to input and lexicalGlobalObject to global.